okta/src/main/java/org/ccci/idm/user/okta/dao/OktaUserDao.kt

```diff
@@ -99,9 +104,10 @@ class OktaUserDao(private val okta: Client, private val listeners: List<Listener
         val oktaGroup = group.id?.let { okta.getGroup(it) } ?: throw GroupNotFoundException()
 
         return oktaGroup.listUsers().stream()
-            .filter { it.matches(expression) }
-            .restrictMaxAllowed(restrictMaxAllowed)
             .map { it.asIdmUser(loadGroups = false) }
+            .filter { !it.isDeactivated || includeDeactivated }
+            .filter { expression?.matches(it) != false }
```

Comment:
maybe simplify by dropping ` != false `

Reply:
This is actually a Kotlin idiom because the statement could evaluate to null. So this is the recommended way of handling it